File: strings/3. Longest Substring Without Repeating Characters/lengthOfLongestSubstring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_set = set()
        start = 0
        result = 0
        
        for end in range(len(s)):
            while s[end] in char_set:
                char_set.remove(s[start])
                start += 1
            char_set.add(s[end])
            result = max(result, end - start + 1)
        
        return result

File: strings/3. Longest Substring Without Repeating Characters/lengthOfLongestSubstring.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let charSet = new Set();
    let start = 0;
    let result = 0;

    for (let end = 0; end < s.length; end++) {
        while (charSet.has(s[end])) {
            charSet.delete(s[start]);
            start++;
        }
        charSet.add(s[end]);
        result = Math.max(result, end - start + 1);
    }

    return result;
};